kpops/components/base_components/base_defaults_component.py

```diff
@@ -17,12 +17,41 @@
 
 
 class BaseDefaultsComponent(BaseModel):
-    type: str = Field(default=..., const=True)
+    """Base for all components, handles defaults.
+
+    Component defaults are usually provided in a yaml file called
+    `defaults.yaml`. This class ensures that the defaults are read and assigned
+    correctly to the component.
+
+    :param type: Component type
+    :type type: str
+    :param enrich: Whether to enrich component with defaults, defaults to False
+    :type enrich: bool, optional
+    :param config: Pipeline configuration to be accessed by this component
+    :type config: PipelineConfig
+    :param handlers: Component handlers to be accessed by this component
+    :type handlers: ComponentHandlers
+    """
+
+    type: str = Field(default=..., description="Component type", const=True)
 
-    enrich: bool = Field(default=False, exclude=True, hidden_from_schema=True)
-    config: PipelineConfig = Field(default=..., exclude=True, hidden_from_schema=True)
+    enrich: bool = Field(
+        default=False,
+        description="Whether to enrich component with defaults",
+        exclude=True,
+        hidden_from_schema=True,
+    )
+    config: PipelineConfig = Field(
+        default=...,
+        description="Pipeline configuration to be accessed by this component",
```

Comment:
it doesn't hurt to have a description here, but it wouldn't be required since the `config` and `handlers` attributes aren't part of the generated pipeline schema

Reply:
Yeah, all class attributes in `BaseDefaultsComponent` are in the same position. I added the descriptions because it seemed like a way to document them, but It could be seen as kinda redundant as I have written the same descriptions in the class docstring right above.

A third option is adding a docstring below the assignments as outlined in [PEP 257](https://peps.python.org/pep-0257/#what-is-a-docstring), but I don't think it is needed.

Do you think that I should remove the descriptions from the assignments?